stripe-node/app.js

```diff
@@ -52,31 +50,53 @@ app.post('/charge/angular', (req, res) => {
     source: req.body.token.id,
     shipping: address,
     name: req.body.token.name,
-  })
-  // Add charge to customer
-    .then(customer => stripe.charges.create({
+  }, (err, customers) => {
+    stripe.charges.create({
       amount: req.body.amount,
       currency: req.body.currency,
       description: 'Donation to Keyman',
-      customer: customer.id,
+      customer: customers.id,
       receipt_email: req.body.token.email,
-    }));
-// Successfull api call
-  // .then(res => {
-  //   res.status(200)
-  //     .json({
-  //       status: '200',
-  //       message: res,
-  //     });
-  // })
-  // Handle api failure
-  // .catch(res => {
-  //   res.status(500)
-  //      .json({
-  //        status: '500',
-  //        message: res,
-  //      });
-  // })
+    }, (err, charges) => {
+      // Handle errors
+      console.log(err.type);
+      if (err !== null) {
+        switch (err.type) {
+          case 'StripeCardError':
+            // A declined card error
+            err.message = "Your card's expiration year is invalid.";
```

Comment:
A card can be declined for many reasons. A message should be provided from the bank, it seems in the `message` property already?

Actually, I'm not sure why you are not just using the `message` property and instead are overwriting it in the error handler?

Reply:
This is what stripe provided for error handling. See [https://stripe.com/docs/api/errors/handling] I'll see about passing in the error message.

Reply:
So according to the [Stripe docs](https://stripe.com/docs/api/errors#errors-message), `message` is passed back from Stripe and for card errors is appropriate to display to end users. Other messages are apparently not useful for end users.

Reply:
That makes sense. So the user/application won't know if any of the errors listed unless a response is sent back and any error info is helpful. It's better than having no error information at all.

Reply:
I'll go ahead set it back to err.message. Think it would be good to set that for all of them.